### PR TITLE
[XPU] update XPU device info, test=develop

### PR DESCRIPTION
**Pull request by qili93**
### PR types

Others

### PR changes

Others

### Describe

Update XPU device info to support PADDLE_ENFORCE_XPU_SUCCESS and memory copy / device management apis.

### CODE REVIEW

File: paddle/fluid/memory/allocation/naive_best_fit_allocator.cc [hunk 1]
```diff
@@ -205,30 +176,9 @@ void Free<platform::XPUPlace>(const platform::XPUPlace &place, void *p,
...
   VLOG(10) << "Allocate " << size << " bytes on " << platform::Place(place);
   VLOG(10) << "Free pointer=" << p << " on " << platform::Place(place);
-  int dev_id = -1;
-  int ret = xpu_current_device(&dev_id);
-  PADDLE_ENFORCE_EQ(ret, XPU_SUCCESS,
-                    platform::errors::External(
-                        "XPU API return wrong value[%d], please check whether "
-                        "Baidu Kunlun Card is properly installed.",
-                        ret));
-  if (dev_id >= 64) {
```
Comment (QingshuChen) [inline paddle/fluid/memory/allocation/naive_best_fit_allocator.cc:215]: 如果是模拟器，没有-64， 这块调用XPUDeviceGuard是不是就出错了

Comment (qili93) [reply paddle/fluid/memory/allocation/naive_best_fit_allocator.cc:215]: XPUDeviceGuard的定义在xpu_info.h里面，其中调用了`int prev_id = platform::GetXPUCurrentDeviceId();`，这里`GetXPUCurrentDeviceId`的实现在xpu_info.cc里面，代码如下，里面有处理dev_id >= 64的情况

```c++
int GetXPUCurrentDeviceId() {
  int dev_id;
  PADDLE_ENFORCE_XPU_SUCCESS(xpu_current_device(&dev_id));
  if (dev_id >= 64) {
    // if dev_id >= 64, the device is a simulator device, -64 to get real dev_id
    dev_id -= 64;
  }
  return dev_id;
}
```

**Review by QingshuChen — APPROVED** [review]
LGTM

**Review by Shixiaowei02 — APPROVED** [review]

**Review by zhiqiu — APPROVED** [review]
LGTM for memory/*

**Pull request merged**